spec/models/appointment_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.describe Appointment, type: :model do
+  before(:each) do
+    @user = User.create(name: 'John Doe')
+    @doctor = Doctor.create(name: 'Dr. John Doe', specialization: 'Pediatrics')
+  end
+
+  describe 'associations' do
+    it 'should belong to user' do
+      assc = described_class.reflect_on_association(:user)
+      expect(assc.macro).to eq :belongs_to
+    end
+
+    it 'should belong to doctor' do
+      assc = described_class.reflect_on_association(:doctor)
+      expect(assc.macro).to eq :belongs_to
+    end
+  end
+
+  describe 'validations' do
+    it 'is not valid without a description' do
+      appointment = Appointment.new(description: nil)
+      expect(appointment).to_not be_valid
+    end
+
+    it 'is not valid with a description less than 3 characters' do
+      appointment = Appointment.new(description: 'ab')
+      expect(appointment).to_not be_valid
+    end
+
+    it 'is not valid with a description more than 150 characters' do
+      appointment = Appointment.new(description: 'a' * 151)
```

Comment:
For example here : Appointment.new(description: 'test' ) this will never be valid since it does not have a user_id and doctor_id so even if the description is less than 150 characters it will return false.